### PR TITLE
Create Add Sub-command for Event

### LINKED ISSUE #100: Add support for creating events
# 🚀 Feature Request

### Description

Add functionality for creating events.

I am making a big request here and I think that this solution can possibly be applied to other entities. 

### Describe the solution you'd like
Create a new `create` command for the `event` sub-command which creates new events in EdgeX. The issue with events is that it contains a lot of information which we cannot expect the user to provide cleanly via the command line. I think we can use an editor like vi(m) and provide a template for the user to populate which then can be used to create the event. A good example of this is how `git` handles interactive rebase and commit messages(without the `-m`). 

Some other nice feature that would be cool:

1. Allow the user to set their preferred editor in some global way. Like how you can configure it with `git`
2. Allow the user to provide some optional information via flags that will populate some of the fields in the template. This would be useful if someone wants to pipe commands together and use existing information like timestamps.
3. Allow the user to re-edit their changes if there is an error sending the information to EdgeX. For example, if I mistype something which results in a 400(Bad request), it would be nice to be able to fix the issue without re-typing everything especially since piped information can be lost.

### Describe alternatives you've considered
Allow the user to provide the information via flags and/or arguments. I decided to not go this route for ease of use as there is too much information to be provided via CLI.

### PR DESCRIPTION
**Pull request by AnthonyMBonafide**
Fix #100

Create new Add sub-command for Event.

Add functionality for Interactive mode which allows users to provide
information via the editor of their choice.

Updated README to include new Interactive mode functionality.

Signed-off-by: Anthony M. Bonafide <AnthonyMBonafide@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -70,6 +70,20 @@ If your EdgeX instance is not running on localhost, minimally you will need to r
 You can now use the CLI by entering `edgex-cli` anywhere on your machine provided your $GOBIN is on your $PATH.
 
 
+#### Interactive Mode
+
+Some commands leverage interactive-mode which opens an editor and allows you to provide information that would 
+normally be difficult with just command line arguments. For example, creating an Event requires a lot of information,
+also Events contain zero or more readings. Using interactive mode, you can easily create an Event with many readings and 
+customize each reading. You can choose the editor that is used by setting the environment variable `EDITOR`. The default
+editor is `Vi` and is used if no `EDITOR` is specified. Some examples of editors:
```
Comment (mkbhanda) [inline README.md:79]: Could we make the default based on whether the operating system is Linux or Windows. as opposed to always vi ?

**Comment by AnthonyMBonafide** [comment]
@mkbhanda , @difince 

I am not able to add reviewers. I am bringing this up so this does not get lost.

File: cmd/root.go [hunk 2]
```diff
@@ -35,6 +33,9 @@ import (
 	"github.com/edgexfoundry-holding/edgex-cli/cmd/version"
 	"github.com/edgexfoundry-holding/edgex-cli/config"
 	"github.com/edgexfoundry-holding/edgex-cli/pkg/pager"
+
+	"github.com/spf13/cobra"
```
Comment (mkbhanda) [inline cmd/root.go:37]: Good catch!

File: pkg/editior/interactive.go [hunk 3]
```diff
@@ -0,0 +1,82 @@
+/*******************************************************************************
+ * Copyright 2020 Dell Inc.
```
Comment (mkbhanda) [inline pkg/editior/interactive.go:2]: Would you please rename this directory to "editor" versus the current version -- feels like a typo.

**Comment by difince** [comment]
@AnthonyMBonafide Thank you for your contribution! Happy to see you here :)

I really like the idea of having interactive modе, even that the code base will become even more tight with the changes in go-mod-core-contracts. 
We could consider implementing the interactive mode with the other commands as well. 
But overall I think that it is still valuable to keep the option of sending file that contains all the data? 
Creating the first request is always harder. The interactive mode could ease this step. But once you have the template, I guess that editing already existing file and providing only one flag would be easier... 
We should validate how it works on Windows. Overall we need to test how the entire CLI behaves with Windows. I plan to test it next week.
Sorry for the late response these days we got plenty of holidays in Bulgaria. 

LGTM

**Comment by mkbhanda** [comment]
@difince , I like your idea of a template sample file, along the lines of sample/createNotification.toml. @AnthonyMBonafide would you consider adding support for such an option. Your "tight" coupling comment makes me wonder whether we should have a samples/geneva, samples/fuji directory or should they be named samples/1.2/ ... the number saves us having to deal with mapping from release semantic version to names like Fuji, Geneva etc. If we go the Fuji route would need to change the response from .edgex-cli version to return the name too.

File: README.md (hunk 1 above)
Comment (AnthonyMBonafide) [reply README.md:79]: Very good point. What would be a good editor for Windows? Notepad?

**Comment by AnthonyMBonafide** [comment]
@difince, @mkbhanda Thank you

I like the comments you both have provided. I will work on implementing your suggestions.

File: README.md (hunk 1 above)
Comment (mkbhanda) [reply README.md]: You got it, faithful Notepad!

**Pull request merged**